### DISCUSSION TITLE
A  question from Lesson 12: Foundry DeFi | Stablecoin

### DISCUSSION THREAD

**Discussion by omarhibshi** [Cyfrin/foundry-full-course-cu#842] [Q&A]
The below log  is from one of the unit test. At the very end after the liquidation, I see that USER's DSC token balance remains intact whereas LIQUIDATOR's DSC token balance is zero. Shouldn't it be the other way around?

 Prior to collateral token price crash 
  ----------------------------------------------------
  USER DSC balance                          100000000000000000000
  USER weth balance                          0
  USER's TotalDscMinted                    100000000000000000000
  USER's TotalCollateralValueInUsd    20000000000000000000000
  MIN_HEALTH_FACTOR                   1000000000000000000
  USER's HealtheFactor                      100000000000000000000
  
  Post collateral token price crash 
  -----------------------------------------------
  USER DSC balance                           100000000000000000000
  USER weth balance                           0
  USER's TotalDscMinted                     100000000000000000000
  USER's TotalCollateralValueInUsd    180000000000000000000
  MIN_HEALTH_FACTOR                   1000000000000000000
  USER's HealtheFactor                      900000000000000000
  
  ----------------------------------------------------
  liquidator enters, deposits collateral and mints DSC
  ----------------------------------------------------
  liquidator's weth initial balance   20000000000000000000
  
  Pre Liquidation
  ----------------------------------------------------
  LIQUIDATOR DSC balance                   100000000000000000000
  liquidator's weth balance                        0
  liquidator's TotalDscMinted                    100000000000000000000
  liquidator's TotalCollateralValueInUsd    360000000000000000000
  priceFeed after price crash                    1800000000
  Debt 100e18 DSC = ?? weth                 5555555555555555555
  Total Collateral To Redeem                   6111111111111111110
  USER remaining collatreral                  6111111111111111110
  
  Post Liquidation 
  -----------------------------------------------
  USER .....
  USER DSC balance                           100000000000000000000
  USER weth balance                          0
  USER's TotalDscMinted                    0
  USER's TotalCollateralValueInUsd    70000000000000000020
  MIN_HEALTH_FACTOR                   1000000000000000000
  USER's HealtheFactor                           115792089237316195423570985008687907853269984665640564039457584007913129639935
  
  LIQUIDATOR .....
  LIQUIDATOR DSC balance                          0
  LIQUIDATOR weth balance                          6111111111111111110
  LIQUIDATOR's TotalDscMinted                    100000000000000000000
  LIQUIDATOR's TotalCollateralValueInUsd    360000000000000000000
  MIN_HEALTH_FACTOR                               1000000000000000000
  LIQUIDATOR's HealtheFactor                      1800000000000000000

**Comment by shikhar229169** [comment]
Think of DSC as a loan.
The DSC Engine's mapping ```dscMinted``` tracks the DSC loan entries of users, while the Decentralized Stable Coin contract holds the actual balance of users.

So, by liquidation, the liquidator actually paid the loan of user, that's why the user's dscMinted becomes 0 (the loan entry), so someone else paid the user's loan but they still hold the balance. (Suppose I took 10$ from a person, and you paid them back their 10$ on behalf of me, but still I hold the 10$ with me, the loan was paid by you)

And as the liquidator paid the user's loan of dsc, so the liquidator's actual dsc balance gets reduced but they still have a loan.

**Reply by omarhibshi** [reply]
Thanks a lot @shikhar229169 

Your explanation has clarified the point for me quite perfectly.
